src/document.js

```diff
@@ -576,8 +576,9 @@ class DocumentSnapshot {
    * @return {boolean}
    */
   get isEmpty() {
-    return !this._fieldsProto || isEmptyObject(this._fieldsProto);
+    return this._fieldsProto === undefined || isEmptyObject(this._fieldsProto);
```

Comment:
Since we have `is`, how about `is.undefined(this._fieldsProto)`?

Reply:
There is also `is.empty(obj)`

Reply:
Good suggestions. I removed `isEmptyObject`.